utils/build-windows.bat

```diff
@@ -0,0 +1,314 @@
+:: build-windows.bat
+::
+:: This source file is part of the Swift.org open source project
+::
+:: Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+:: Licensed under Apache License v2.0 with Runtime Library Exception
+::
+:: See https://swift.org/LICENSE.txt for license information
+:: See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+
+:: REQUIRED ENVIRONMENT VARIABLES
+:: This script requires to be executed inside one Visual Studio command line,
+:: in order for many of the tools and environment variables to be available.
+:: Additionally, it needs the following variables:
+:: - CMAKE_BUILD_TYPE: Kind of build: Release, RelWithDebInfo, Debug.
+:: - PYTHON_HOME: The Python installation directory.
+
+:: REQUIRED PERMISSIONS
+:: Practically, it is easier to be in the Adminstrators group to run the
+:: script, but it should be possible to execute as a normal user.
+:: The user will need permission to write files into the Windows SDK and the
+:: VisualC++ folder.
+
+@echo off
+
+setlocal enableextensions enabledelayedexpansion
+
+set "exitOnError=|| (exit /b)"
```

Comment:
I'm not sure I buy the value of the `%exitOnError%`

Reply:
`exitOnError` is a “macro”. Every place I write it, it is like I wrote `|| (exit /b)`. So `foobar %exitOnError%` is actually `foobar || (exit /b)`. The `/b` avoids existing the command prompt, and only exiting the batch script. Since the script is written with `call` and `goto :end`, it will finish the subroutine, but the main batch script will continue executing (which is not exactly true, because most `call` are also marked `%exitOnError%`, but it allows for some errors to be ignored, if needed). I don’t provide an explitic error level to not override the last error level.

This is basically like a manual way of writing the typical Bash `set -e`.

Reply:
Right, I understand what it is doing.  The problem is that is is very non-idiomatic.  The idiomatic thing is to expand the macro (`|| (exit /b)` is very common in larger batch files).  That makes it harder for those who are familiar with batch files but not swift to contribute.

Reply:
The benefit of the macro is that it can be disabled with a comment, and the file becomes "continue on error" with a little change. It is very helpful while debugging.

Reply:
I'm not convinced that the tradeoff is worth it.  But, let's get this enabled.